payjoin/src/receive/v2/error.rs

```diff
@@ -61,6 +41,14 @@ impl From<OhttpEncapsulationError> for Error {
     }
 }
 
+impl From<DirectoryResponseError> for Error {
+    fn from(e: DirectoryResponseError) -> Self { InternalSessionError::DirectoryResponse(e).into() }
+}
+
+impl From<DirectoryResponseError> for SessionError {
+    fn from(e: DirectoryResponseError) -> Self { InternalSessionError::DirectoryResponse(e).into() }
+}
+
```

Comment:
Ah, I see this commit obviates my earlier comments. However, it still seems like it depends on context which one of these it gets converted to, which suggests to me you still want a map instead of the conversion (which happens on only few occasions)